engine/runtime-compiler/src/main/java/org/enso/compiler/pass/analyse/types/scope/StaticMethodResolution.java

```diff
@@ -0,0 +1,123 @@
+package org.enso.compiler.pass.analyse.types.scope;
+
+import java.util.Collection;
+import java.util.List;
+import org.enso.compiler.common.MethodResolutionAlgorithm;
+import org.enso.compiler.pass.analyse.types.TypeRepresentation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** The implementation of {@link MethodResolutionAlgorithm} for static analysis. */
+public final class StaticMethodResolution
+    extends MethodResolutionAlgorithm<
+        TypeRepresentation, TypeScopeReference, StaticImportExportScope, StaticModuleScope> {
+  private final ModuleResolver moduleResolver;
+  private final BuiltinsFallbackScope builtinsFallbackScope;
+  private final Logger logger = LoggerFactory.getLogger(this.getClass());
```

Comment:
```suggestion
  private static final Logger logger = LoggerFactory.getLogger(StaticMethodResolution.class);
```
also Java guidelines would suggest to rename to `LOGGER`.

Reply:
Thanks for this and the related suggestions, I somehow missed that. I will rename them.